src/Groups/schur_index.jl

```diff
@@ -0,0 +1,520 @@
+    _, a, b = gcdx(ppart, m)
+    r = m == 1 ? 1 : modord(p, m)
+#T `modord(N, 1)` could return 1 (smallest pos. int. i s.t. 1 divides (N^i-1))
+    stab = Int[]
+    sigma_p = mod(1 + a*(p-1)*ppart, N)
+    res = ppart == 1 ? [1] : Vector{Int}(GAP.Globals.PrimeResidues(GapObj(ppart)))
+    for u in res
+      for i in 0:(r-1)
+        sigma = mod(sigma_p^i * (1 + (u-1)*m*b), N)
+        if GAP.Globals.GaloisCyc(vals, GapObj(sigma)) == vals
+          push!(stab, sigma)
+        end
+      end
+    end
+
+    return (N, stab, r*euler_phi(ppart))
+  end
+end
+
+
+# `elm` is an element of a cyclotomic field in GAP.
+# `info` is a description `(N, stab, index)` of the field
+# $F \subseteq \Q_p(N)$ such that all elements in $F$ are fixed under the
+# Galois automorphisms $*k$, for $k \in stab$.
+# (We have $[\Q_p(N):F] = index$, but this is nnot used here.)
+function _membership_test(elm::GAP.Obj, info::Tuple{Int, Vector{Int}, Int})
+  return mod(info[1], GAPWrap.Conductor(elm)) == 0 &&
+         all(k -> GAP.Globals.GaloisCyc(elm, k) == elm, info[2])
+end
+
+
+#############################################################################
+
+"""
+    local_schur_indices(chi::GAPGroupClassFunction; cyclic_defect::Vector{Int} = Int[])
+
+Return the array of pairs `p => m_p(chi)` such that `m_p(chi) > 1` holds.
+The Schur index over the real field is stored at `p = -1` if applicable.
+
+`cyclic_defect` contains primes `p` such that `chi` is known to belong to
+a `p`-block of cyclic defect.
+
+# Examples
+```jldoctest
+julia> G, chi = Oscar.yamada_example(5, 2);
+
+julia> order(G)
+40
+
+julia> local_schur_indices(chi)
+2-element Vector{Pair{Int64, Int64}}:
+ -1 => 2
+  5 => 2
+
+julia> schur_index(chi)
+2
+```
+"""
+function local_schur_indices(chi::GAPGroupClassFunction; cyclic_defect::Vector{Int} = Int[])
+  @assert characteristic(chi) == 0
+  tbl = parent(chi)
+  deg = degree(ZZRingElem, chi)
+  chipos = findfirst(isequal(chi), tbl)
+  @assert chipos !== nothing "chi must be irreducible"
+  @vprintln :SchurIndices 1 "called with character of degree $deg"
+  G = group(tbl)
+  result = Pair{Int, Int}[]
+
+  # Step 1:
+  # Let `m` denote the Schur index.
+  # Determine m_\R(chi) and use Theorem 2.2 to find a bound u \in \N
+  # such that m(chi) divides u.
+  # If u = 1 terminate.
+  # `m` divides the `chi(1)`.
+  ind = indicator(chi)
+  if ind == 0
+    m_R = 1
+  elseif ind == -1
+    m_R = 2
+    push!(result, -1 => 2)
+  else
+    m_R = 1
+  end
+
+  u = deg
+  # `m` divides `|G|/chi(1)`.
+  Gorder = order(tbl)
+  u = gcd(u, div(Gorder, deg))
+  if u == 1
+    @vprintln :SchurIndices 1 "m = 1 (divides chi(1) and |G|/chi(1))"
+    return result
+  end
+
+  # The character field contains a primitive `m`-th root of unity.
+  if ind == 0
+    # Compute the conductor of the largest cyclotomic field
+    # that is contained in the character field of `chi`.
+    gapfield = GAPWrap.Field(GapObj(chi))
+    N = GAPWrap.Conductor(gapfield)
```

Comment:
I was confused by this because I read "Compute the conductor" as referring to the value `N`... but I think it's really talking about `cond`. How about this to make it clearer?
```suggestion
    # Compute the conductor `N` of the smallest cyclotomic field
    # that contains the character field of `chi`.
    gapfield = GAPWrap.Field(GapObj(chi))
    N = GAPWrap.Conductor(gapfield)
    # Compute the conductor `cond` of the largest cyclotomic field
    # that is contained in the character field of `chi`.
```